bobikApp/TableVC.swift

```diff
@@ -0,0 +1,126 @@
+        return people.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellID, for: indexPath) as! CustomCell
+        //cell.nameLabel.text = people[indexPath.row].name
+        if openedCellIndex == indexPath.row {
+            cell.bioLabel.text = people[indexPath.row].bio
+        } else {
+            cell.bioLabel.text = "tap to expand -->"
+        }
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if openedCellIndex == indexPath.row {
+            let tappedCell = tableView.cellForRow(at: indexPath) as! CustomCell
+            return
+        } else if openedCellIndex == nil {
+            guard let tappedCell = tableView.cellForRow(at: indexPath) as? CustomCell else {return}
+            tappedCell.nameTextView.isUserInteractionEnabled = true
+            tappedCell.nameTextView.resignFirstResponder()
+            tappedCell.bioLabel.text = people[indexPath.row].bio
+            tableView.beginUpdates()
+            tableView.endUpdates()
+            openedCellIndex = indexPath.row
+            return
+        } else if let openedCell = openedCellIndex {
+            let indexPath = IndexPath(row: openedCell, section: (tableView.numberOfSections - 1))
+            if let lastTappedCell = tableView.cellForRow(at: indexPath) as? CustomCell {
+                lastTappedCell.nameTextView.isUserInteractionEnabled = false
+                lastTappedCell.bioLabel.text = "tap to expand -->"
+                tableView.beginUpdates()
+                tableView.endUpdates()
+            }
+            openedCellIndex = nil
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        people.remove(at: indexPath.row)
+        if openedCellIndex == indexPath.row {
+            openedCellIndex = nil
+        }
+        let rowIndex = [indexPath]
+        tableView.deleteRows(at: rowIndex, with: .automatic)
+    }
+
+    func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldReceive touch: UITouch) -> Bool {
+        return (touch.view === tableView)
+    }
+    
+    @objc func closeTheCell(_ gesture: UITapGestureRecognizer) {
+        if let openedCell = openedCellIndex {
+            let indexPath = IndexPath(row: openedCell, section: tableView.numberOfSections - 1)
+            let cell = tableView.cellForRow(at: indexPath) as! CustomCell
+            cell.nameTextView.isUserInteractionEnabled = false
+            cell.nameTextView.resignFirstResponder()
+            cell.bioLabel.text = "tap to expand -->"
+            tableView.beginUpdates()
+            tableView.endUpdates()
+            openedCellIndex = nil
+        }
+    }
```

Comment:
Right now you're configuring the cell in three different places, which complicates the logic. You could think about unifying the current behavior. For example, add a level of abstraction, when you click on a cell or in some place you call a method of configuring a cell, which will decide how to display the cell.

Reply:
Right now you have some part of the cell configuation being customized in cellForRow, another part when you click on the cell, one more part when you click on an empty space. As the application gets more complex, this will lead to bugs.

Reply:
partially done I guess